### PR TITLE
feat: add path to claim "user roles" in app settings

### PR DESCRIPTION
**Pull request by astsiapanay**
We need to support access to the user roles claim in JWT token for multiple Identity Providers.
New setting "rolePath" allows the app to read user roles from JWT.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -27,26 +27,26 @@ Static settings are used on startup and cannot be changed while application is r
...
-|storage.identity                              |-                          |Blob storage access key
-|storage.credential                            |-                          |Blob storage secret key
-|storage.bucket                                |-                          |Blob storage bucket
-|storage.createBucket                          |false                      |Indicates whether bucket should be created on start-up
+| Setting                              |Default                    |Description
+|--------------------------------------|-|-
+| config.files                         |aidial.config.json         |Config files with parts of the whole config.
+| config.reload                        |60000                      |Config reload interval in milliseconds.
+| identityProvider.jwksUrl             |-                          |Url to jwks provider.
+| identityProvider.rolePath            |dial                       |Path to the claim user roles in JWT token.
```
Comment (Maxim-Gadalov) [inline README.md]: please add an example how to construct path

File: src/main/java/com/epam/aidial/core/security/IdentityProvider.java [hunk 2]
```diff
@@ -23,15 +25,17 @@
 import java.util.Objects;
 import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.TimeUnit;
+import java.util.function.Function;
+import java.util.function.Supplier;
```
Comment (Maxim-Gadalov) [inline src/main/java/com/epam/aidial/core/security/IdentityProvider.java]: please remove unused imports

File: src/main/java/com/epam/aidial/core/security/IdentityProvider.java [hunk 3]
```diff
@@ -93,16 +93,29 @@ private void evictExpiredJwks() {
...
-        if (resourceAccess == null) {
-            return Collections.emptyList();
+        if (rolePath.length == 1) {
+            return token.getClaim(rolePath[0]).asList(String.class);
         }
-        Map<String, Object> app = (Map<String, Object>) resourceAccess.get(appName);
-        if (app == null) {
-            return Collections.emptyList();
+        Map<String, Object> claim = token.getClaim(rolePath[0]).asMap();
+        List<String> defaultResult = Collections.emptyList();
```
Comment (Maxim-Gadalov) [inline src/main/java/com/epam/aidial/core/security/IdentityProvider.java]: consider moving this to the constant

Comment (astsiapanay) [reply src/main/java/com/epam/aidial/core/security/IdentityProvider.java]: fixed

File: src/main/java/com/epam/aidial/core/security/IdentityProvider.java (hunk 2 above)
Comment (astsiapanay) [reply src/main/java/com/epam/aidial/core/security/IdentityProvider.java]: fixed

File: README.md (hunk 1 above)
Comment (astsiapanay) [reply README.md]: fixed

**Review by Maxim-Gadalov — APPROVED** [review]

**Pull request merged**